src/plonk/circuit.rs

```diff
@@ -1102,7 +1083,46 @@ impl<F: Field> ConstraintSystem<F> {
                 .unwrap_or(0),
         );
 
-        degree
+        std::cmp::max(degree, self.minimum_degree.unwrap_or(1))
+    }
+
+    /// Compute the number of blinding factors necessary to perfectly blind
+    /// each of the prover's witness polynomials.
+    pub fn blinding_factors(&self) -> usize {
+        // All of the prover's advice columns are evaluated at no more than
+        let factors = *self.num_advice_queries.iter().max().unwrap_or(&1);
+        // distinct points during gate checks.
+
+        // In the permutation and lookup argument the witness polynomials are
+        // evaluated at most 3 times:
```

Comment:
If I'm reading things correctly, the polynomials for each lookup argument are evaluated at most 2 times:
- `product_poly` at `cur` and `next`
- `permuted_input_poly` at `cur` and `prev`
- `permuted_table_poly` at `cur`

Meanwhile for the permutation argument, we evaluate `permutation_product_poly` at `cur`, `next`, and (for all except the final set) `last`.

So it would be somewhat clearer here to say:
```suggestion
        // - The permutation argument witness polynomials are evaluated at most 3 times.
        // - Each lookup argument has independent witness polynomials, and they are
        //   evaluated at most 2 times.
```